src/type2-extension.c

```diff
@@ -0,0 +1,113 @@
+#include "vctrs.h"
```

Comment:
I imagine that very soon it will also hold `date_ptype2()` and extensions for bare tibbles too.

I could keep `vec_ptype2_extension()` here, but move `fct_ptype2()` and `ord_ptype2()` to a `type-factor.c` file? Then we’d eventually get a `type-date.c` as well

Reply:
yes that would mirror the .R files nicely.